stl/inc/atomic

```diff
@@ -2346,12 +2346,7 @@ private:
     using _Base = _Choose_atomic_base_t<_Ty, _Ty&>;
 
 public:
-    // clang-format off
-    static_assert(is_trivially_copyable_v<_Ty> && is_copy_constructible_v<_Ty> && is_move_constructible_v<_Ty>
-        && is_copy_assignable_v<_Ty> && is_move_assignable_v<_Ty>,
-        "atomic_ref<T> requires T to be trivially copyable, copy constructible, move constructible, copy assignable, "
-        "and move assignable.");
-    // clang-format on
+    static_assert(is_trivially_copyable_v<_Ty>, "atomic_ref<T> requires T to be trivially copyable.");

```

Comment:
We were intentionally strengthening the class requirement to deal with the fact that the design never considered cv-qualified types. (I have a TODO to file an LWG issue for this, which I should have made a GitHub issue.) This isn't a sufficient condition to allow `store` to work with such types, for example, so `atomic_ref<const int>` is unimplementable as specified. If we're going to remove requirements here in an attempt to allow cv-qualified types, we should determine what the proper requirements are for each operation and submit an LWG issue to fix the class template.

Reply:
I think the purpose of `atomic_ref<const T>` is to only work with `load` and `wait` like `const atomic<T>`

Reply:
> I think the purpose of `atomic_ref<const T>` is to only work with `load` and `wait` like `const atomic<T>`

I can also envision an alternative specification that makes sense. It's our responsibility as implementors to either implement the specification in the working draft as-is, or submit defect reports to fix the specification to agree with what we implement.

Reply:
@StephanTLavavej and I discussed this PR off-line, and decided the best way forward is to assume that the Standard intends for cv-qualified types to work, despite failing to account for them in the wording. As such we'll implement the things that are implementable as specified and avoid the things that aren't implementable as specified. I'm to file an LWG issue about the specification issues and we'll move forward with this PR.